app/javascript/dashboard/routes/dashboard/settings/personal/MessageSignature.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+  <div class="flex flex-col gap-6">
+    <form class="flex flex-col gap-6" @submit.prevent="updateSignature()">
+      <woot-message-editor
+        id="message-signature-input"
+        v-model="signature"
+        class="message-editor h-[10rem] !px-3"
+        :is-format-mode="true"
+        :placeholder="$t('PROFILE_SETTINGS.FORM.MESSAGE_SIGNATURE.PLACEHOLDER')"
+        :enabled-menu-options="customEditorMenuList"
+        :enable-suggestions="false"
+        :show-image-resize-toolbar="true"
+      />
+      <form-button
+        type="submit"
+        color-scheme="primary"
+        variant="solid"
+        size="large"
+      >
+        {{ $t('PROFILE_SETTINGS.FORM.MESSAGE_SIGNATURE_SECTION.BTN_TEXT') }}
+      </form-button>
+    </form>
+  </div>
+</template>
+<script setup>
+import { defineProps, ref, watch } from 'vue';
+import WootMessageEditor from 'dashboard/components/widgets/WootWriter/Editor.vue';
+import { MESSAGE_SIGNATURE_EDITOR_MENU_OPTIONS } from 'dashboard/constants/editor';
+import FormButton from 'v3/components/Form/Button.vue';
+
+const props = defineProps({
+  messageSignature: {
+    type: String,
+    default: '',
+  },
+});
+
+const customEditorMenuList = MESSAGE_SIGNATURE_EDITOR_MENU_OPTIONS;
+const signature = ref(props.messageSignature);
+const emit = defineEmits(['update-signature']);
+
+watch(
+  () => props.messageSignature,
+  newValue => {
+    signature.value = newValue;
+  }
+);
+
+const updateSignature = () => {
+  emit('update-signature', { signature: signature.value }, 'signature');
+};
```

Comment:
Here, a computed getter and setter would work much better

```
const signatureComputed = computed({
  get: () => props.messageSignature,
  set: (newValue) => {
    emit('update-signature', { signature: newValue }, 'signature');
  },
});
```

The current approach introduces a mutable state `signature` while, `signatureComputed` is not. Also the reactivity in the previous case came from the `watch`, here the reactivity is handled by vue since `signatureComputed` is a computedRef.

Reply:
If we use computed the `update-signature` function is called on every input, which is not necessary. We only need to save the signature when the user clicks the "Save message signature" button.

Reply:
@scmmishra

Reply:
Makes sense, resolving this